api/test/e2enew/base/http.go

```diff
@@ -92,3 +104,44 @@ func BatchTestServerPort(times int) map[string]int {
 
 	return res
 }
+
+func GetReader(reqParams map[string]string, contentType string, files []UploadFile) (io.Reader, string, error) {
+	if strings.Index(contentType, "json") > -1 {
+		bytesData, _ := json.Marshal(reqParams)
+		return bytes.NewReader(bytesData), contentType, nil
+	}
+	if files != nil {
+		body := &bytes.Buffer{}
+		writer := multipart.NewWriter(body)
+		for _, uploadFile := range files {
+			file, err := os.Open(uploadFile.Filepath)
+			if err != nil {
+				return nil, "", err
+			}
+			part, err := writer.CreateFormFile(uploadFile.Name, filepath.Base(uploadFile.Filepath))
+			if err != nil {
+				return nil, "", err
+			}
+			_, err = io.Copy(part, file)
+			defer file.Close()
```

Comment:
`defer file.Close()` should be put when the file is opened. At line 121.
Else if the writer.CreateFormFile's err is not nil, the file wouldn't be closed. This will cause `fd leak`.

Reply:
Yes, this is my negligence.
Done.
Thks.